src/components/activityEventElems/ActivityElement/ActivityElement.tsx

```diff
@@ -87,27 +124,177 @@ function Subject({ subject }: { subject: string | JSX.Element | null }) {
   )
 }
 
-export function ActivityEvent({
+function _ActivityEvent({
   header,
   subject,
-  event,
+  details,
   extra,
+  style,
 }: {
-  header: string
+  header: string | JSX.Element | null
   subject: string | JSX.Element | null
   extra?: string | JSX.Element | null
-  event: ActivityElementEvent
+  details: Partial<ActivityElementEvent>
+  style?: CSSProperties
 }) {
   return (
-    <>
+    <div style={style}>
       <DetailsContainer>
         <div>
           <Header header={header} />
           <Subject subject={subject} />
         </div>
-        <SideDetails event={event} />
+        <SideDetails details={details} />
       </DetailsContainer>
       {extra ? <ExtraContainer>{extra}</ExtraContainer> : null}
-    </>
+    </div>
+  )
+}
+
+export function ActivityEvent({
+  event,
+  style,
+}: {
+  event: ProjectEvent
+  style?: CSSProperties
+}) {
+  // TODO: we should use v2 where we can, or a better context
+  const { tokenSymbol } = useContext(V1ProjectContext)
+
+  const { Header, Subject, Extra, details } = useMemo(() => {
+    if (event.payEvent) {
```

Comment:
wdyt about one more layer of organization... instead of importing a bunch of Header/Subject/Extra components here, we could replace this with some functions like `if (event.payEvent) return activityElemForPayEvent(event.payEvent)` where... `payEventElem()` returns a `{Header, Subject, Extra}` type

then the individual Header/Subject/Extra components could be contained in those files with only the main `activityElemFor...` function needing to be exported